### PR TITLE
 Correct unused parameter warnings in transform and unary ops

### PR DESCRIPTION
**Pull request by robertmaynard**
Starting in CUDA 11.3, nvcc will start to unconditionally warn about unused parameters on functions/methods that are in anonymous namespaces.

This corrects issues found by this warning in transform and unary op algorithms

### CODE REVIEW

File: cpp/src/transform/row_bit_count.cu [hunk 1]
```diff
@@ -257,18 +257,11 @@ struct flatten_functor {
...
-                  hierarchy_info& h_info,
-                  rmm::cuda_stream_view stream,
-                  size_type cur_depth,
-                  size_type cur_branch_depth,
-                  thrust::optional<int> parent_index)
+  template <typename T, typename... Args>
+  std::enable_if_t<!cudf::is_fixed_width<T>() && !std::is_same<T, string_view>::value &&
+                     !std::is_same<T, list_view>::value && !std::is_same<T, struct_view>::value,
+                   void>
+  operator()(Args&&...)
```
Comment (codereport) [inline cpp/src/transform/row_bit_count.cu:264]: :heart:

**Review by codereport — APPROVED** [review]
lgtm

**Comment by robertmaynard** [comment]
rerun tests

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**